File: regular_assignments/day_027/a_b_and_modulo/solution.rb
class Solution
  def call
    a = 5
    b = 7
    puts "a: #{a}, b: #{b}, highest mod: #{solve(a,b)}"
    a = 30
    b = 7
    puts "a: #{a}, b: #{b}, highest mod: #{solve(a,b)}"
    a = 61
    b = 5
    puts "a: #{a}, b: #{b}, highest mod: #{solve(a,b)}"
    a = 6816621
    b = 8157697
    puts "a: #{a}, b: #{b}, highest mod: #{solve(a,b)}"
  end

  def solve(a, b)
    mod = 0
    if a > b
      mod = a - b
    else
      mod = b - a
    end
    if calculate_mod(a, mod) == calculate_mod(b, mod)
      return mod
    end
    return 0
  end

  def calculate_mod(a, b)
    if a >= 0
      while a >= b
        a -= b
      end
      return a
    else
      while a < b
        a += b
      end
      return a - b
    end
  end
end

File: regular_assignments/day_027/a_b_and_modulo/solution.go
package aBAndModulo

import "fmt"

func Call() {
	a := 5
	b := 7
	fmt.Printf("a: %v, b: %v, highest mod: %v\n", a, b, solve(a, b))
	a = 30
	b = 7
	fmt.Printf("a: %v, b: %v, highest mod: %v\n", a, b, solve(a, b))
	a = 61
	b = 5
	fmt.Printf("a: %v, b: %v, highest mod: %v\n", a, b, solve(a, b))
	a = 6816621
	b = 8157697
	fmt.Printf("a: %v, b: %v, highest mod: %v\n", a, b, solve(a, b))
}

func solve(a, b int) int {
	mod := 0
	if a > b {
		mod = a - b
	} else {
		mod = b - a
	}
	if calculateMod(a, mod) == calculateMod(b, mod) {
		return mod
	}
	return 0
}

func calculateMod(a, b int) int {
	if a >= 0 {
		for a >= b {
			a -= b
		}
		return a
	} else {
		for a < b {
			a += b
		}
		return a - b
	}
}
